ipn/ipnlocal/local.go

```diff
@@ -870,6 +871,8 @@ func (b *LocalBackend) setClientStatus(st controlclient.Status) {
 				b.logf("Failed to save new controlclient state: %v", err)
 			}
 		}
+		b.writeServerModeStartState(userID, prefs.View())
```

Comment:
so this is writing to disk in `setClientStatus`, which is basically the pubsub listener of state changes, but requires a bunch of stuff be up and working. So this works, but it's a little late.

All mutations to the unattended bit etc has to come through LocalBackend anyway, so we should be writing to the state store in e.g. SetPrefs, EditPrefs, Start.

(And sorry, you've found what's probably the oldest & worst code in the tree... we've been slowing moving towards EditPrefs and a new generation of APIs that are more explicit and documented but LocalBackend still deals with a bunch of styles.)

Looks like we should probably pull out all the assignments to `b.prefs` into a `setPrefsLocked` and place all the code that needs to react to prefs changes in one place.

e.g. there's a bunch of this sprinkled all over too:

```
	b.setAtomicValuesFromPrefs(b.prefs)
	b.setTCPPortsInterceptedFromNetmapAndPrefsLocked()
```

... whenever `b.prefs` is assigned.

Reply:
Oh, but it's trying to not do I/O with b.mu held, huh?

I guess setClientStatus (being async) is safer in that regard.

Eh, let's do this for now. Cleaning up LocalBackend is its own project.